src/logger/utils.js

```diff
@@ -0,0 +1,41 @@
+const { Logger, transports, config } = require('winston');
+const { isProd } = require('../utils');
+
+const formatMessage = ({ level, message, meta = {}, colorize }) => {
+  let result = colorize ? config.colorize(level) : level;
+
+  // Prepend the request ID to the message
+  // eslint-disable-next-line newline-after-var
+  let messageStr = message;
+  if (meta.requestId) {
+    messageStr = `[${meta.requestId}] ${messageStr}`;
+  }
+
+  result += ` ${messageStr}`;
+
+  // Add the metadata (if any), but omit the request ID
+  // eslint-disable-next-line newline-after-var
+  const otherMeta = { ...meta };
+  delete otherMeta.requestId;
+
+  if (Object.entries(otherMeta).length) {
+    result += ` ${JSON.stringify(otherMeta)}`;
+  }
+
+  return result;
+};
+
+const createLogger = options => {
+  return new (Logger)({
```

Comment:
Kool, that would work. Though it seems creating a logger and transport instance per request could be a bit of an overhead. Can we reuse the logger instance? A child logger is probably cheaper: https://github.com/winstonjs/winston#creating-child-loggers

Reply:
Thanks for the hint, this looks like the right solution for the problem. I upgraded the code to use child loggers (had to move to Winston 3 though).